Server/VikingVault.DataAccess/Models/AuthorizeUser.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace VikingVault.DataAccess.Models
+{
+
+    public class AuthorizeUser : Attribute, IAuthorizationFilter
+    {
+        void IAuthorizationFilter.OnAuthorization(AuthorizationFilterContext context)
+        {
+            try
+            {
+                string token = context.HttpContext.Request.Headers["x-access-token"].ToString();
+                var tokenObject = new JwtSecurityToken(token);
+                string userId = tokenObject.Payload["Id"].ToString();
+                if (userId == null)
+                {
+                    context.Result = new ForbidResult();
+                }
+            }
+            catch (Exception ex)
```

Comment:
in case of an exception do not return a ForbidResult, return a result with a status code 500, because maybe the exception has nothing to do with authorization

Reply:
It has been changed to return 500 status code